<a href="https://colab.research.google.com/github/kousibrout2006-cell/-Fish-Weight-Prediction/blob/main/Fish_Weight_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Fish Weight Prediction Using Linear Regression

## YBI Foundation Internship Project

### Project Duration
45-Day Business and Data Analyst Internship

### Project Objective

The objective of this project is to predict the weight of a fish
based on its physical characteristics using a Linear Regression
machine learning model.

The dataset contains information about different fish species
and their physical measurements such as length, height, and width.

### Target Variable

Weight

### Features

- Species
- Length1
- Length2
- Length3
- Height
- Width

### Machine Learning Algorithm

Linear Regression

### Tools and Technologies

- Python
- Google Colab
- Pandas
- NumPy
- Matplotlib
- Seaborn
- Scikit-learn
- GitHub

## 1. Project Overview

Fish weight is an important characteristic that can be estimated
from measurable physical characteristics of a fish.

In this project, we will analyze a fish dataset and investigate
the relationship between fish weight and characteristics such as
length, height, and width.

After performing data cleaning and exploratory data analysis,
a Linear Regression model will be developed to predict fish weight.

The performance of the model will be evaluated using appropriate
regression evaluation metrics.

## 2. Problem Statement

The objective is to develop a machine learning model that can
predict the weight of a fish using its physical characteristics.

The model will use features such as:

- Species
- Vertical length (Length1)
- Diagonal length (Length2)
- Cross length (Length3)
- Height
- Width

The predicted weight will then be compared with the actual weight
to evaluate the performance of the Linear Regression model.

## 3. Dataset Description

The dataset contains information about different fish species
and their physical measurements.

### Variables

| Variable | Description |
|----------|-------------|
| Species | Species name of the fish |
| Weight | Weight of fish in grams |
| Length1 | Vertical length in cm |
| Length2 | Diagonal length in cm |
| Length3 | Cross length in cm |
| Height | Height of fish in cm |
| Width | Diagonal width in cm |

### Target Variable

Weight

### Independent Variables

Species, Length1, Length2, Length3, Height, Width

In [ ]:
print("Fish Weight Prediction Project")
print("YBI Foundation Internship")
print("Day 1 - Project Setup")

Fish Weight Prediction Project
YBI Foundation Internship
Day 1 - Project Setup


In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

print("Libraries imported successfully!")

Libraries imported successfully!


## 4. Day 1 Progress

Today, the project environment was successfully established.

### Completed Tasks

- Created GitHub repository
- Created Google Colab notebook
- Defined the project objective
- Defined the problem statement
- Documented the dataset variables
- Identified the target variable
- Identified the independent variables
- Imported the required Python libraries

### Next Step

The next step is to obtain the fish dataset and begin understanding
its structure and contents.

In [5]:
import pandas as pd

url = "https://gist.githubusercontent.com/5chang2/6923330a794da4f4e05c06599c592914/raw/8ffaa770a44541cded1146b524820ef6c50856ec/fish.csv"

df = pd.read_csv(url)

df.head()

,Species,Weight,Length1,Length2,Length3,Height,Width
0,Bream,242.0,23.2,25.4,30.0,11.5200,4.0200
1,Bream,290.0,24.0,26.3,31.2,12.4800,4.3056
2,Bream,340.0,23.9,26.5,31.1,12.3778,4.6961
3,Bream,363.0,26.3,29.0,33.5,12.7300,4.4555
4,Bream,430.0,26.5,29.0,34.0,12.4440,5.1340


In [6]:
print("Number of rows:", df.shape[0])
print("Number of columns:", df.shape[1])

Number of rows: 159
Number of columns: 7


In [7]:
print(df.columns.tolist())


['Species', 'Weight', 'Length1', 'Length2', 'Length3', 'Height', 'Width']


In [8]:
df.tail()

,Species,Weight,Length1,Length2,Length3,Height,Width
154,Smelt,12.2,11.5,12.2,13.4,2.0904,1.3936
155,Smelt,13.4,11.7,12.4,13.5,2.4300,1.2690
156,Smelt,12.2,12.1,13.0,13.8,2.2770,1.2558
157,Smelt,19.7,13.2,14.3,15.2,2.8728,2.0672
158,Smelt,19.9,13.8,15.0,16.2,2.9322,1.8792


In [9]:
df.to_csv("Fish.csv", index=False)

print("Fish.csv created successfully!")

Fish.csv created successfully!


In [10]:
import os

print(os.listdir())

['.config', 'Fish.csv', 'sample_data']


In [11]:
from google.colab import files

files.download("Fish.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# Day 2 - Dataset Collection and Loading

## Objective

The objective of Day 2 was to obtain the Fish Market dataset and
load it into the Google Colab environment for further analysis.

## Dataset

The dataset contains information about different fish species,
their weight, and physical measurements.

## Dataset Variables

- Species - Name of the fish species
- Weight - Weight of the fish in grams
- Length1 - Vertical length in cm
- Length2 - Diagonal length in cm
- Length3 - Cross length in cm
- Height - Height of the fish in cm
- Width - Width of the fish in cm

## Initial Dataset Information

The dataset contains 159 observations and 7 variables.

The target variable for this project is Weight.

The remaining variables will be used as features for predicting
fish weight.

## Day 2 Conclusion

The Fish dataset was successfully obtained and loaded into
Google Colab using Pandas.

The dataset structure and column names were checked, and a local
copy named Fish.csv was created.

Further exploration and analysis will be performed in the next
stage of the project.